.github/workflows/staging.yml

```diff
@@ -0,0 +1,93 @@
+name: Staging
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ develop ]
```

Comment:
leave master here. Any push to master should run the CI. 
This one doesn't do just "staging". This CI compiles the code on every pull-request and push to develop & master. Keep master in both - push and develop